### PR TITLE
Rename fold constructors in Data.Fold 

### LINKED ISSUE #1016: Rename fold constructors in Data.Fold to make them consistent with Streamly.Prelude
```
mkFoldl => foldl'
mkFoldlM => foldlM'
mkFoldr => foldr
```

However, `fold` in `Streamly.Prelude` would still remain inconsistent because it is used to run a fold rather than creating a fold, for creation we would still have `mkFold` for the general fold construction case.

### PR DESCRIPTION
**Pull request by rnjtranjan**
Fix #1016

### CODE REVIEW

File: src/Streamly/Data/Fold.hs [hunk 1]
```diff
@@ -261,3 +261,15 @@ import Prelude
                span, splitAt, break, mapM)
 
 import Streamly.Internal.Data.Fold
+
+{-# INLINE foldl' #-}
+foldl' :: Monad m => (b -> a -> b) -> b -> Fold m a b
+foldl' = mkFoldl
```
Comment (harendra-kumar) [inline src/Streamly/Data/Fold.hs:267]: We should rename `mkFoldl` itself rather than defining a new function.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: src/Streamly/Internal/Data/Fold.hs [hunk 2]
```diff
@@ -19,10 +19,10 @@ module Streamly.Internal.Data.Fold
...
 
     -- * Constructors
-    , mkFoldl
-    , mkFoldlM
+    , foldl'
+    , foldlM'
     , mkFoldl1
-    , mkFoldr
+    , foldr
     , mkFoldrM
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Fold.hs:26]: We can change `mkFoldl1` to `foldl1` and `mkFoldrM` to `foldrM` at the same time. Only `mkFold` and `mkFold_` should remain unchanged.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by adithyaov — APPROVED** [review]

**Comment by adithyaov** [comment]
@rnjtranjan Can you fix up the initial commits?

File: src/Streamly/Internal/Data/Fold.hs [hunk 3]
```diff
@@ -19,11 +19,11 @@ module Streamly.Internal.Data.Fold
...
 
     -- * Constructors
-    , mkFoldl
-    , mkFoldlM
-    , mkFoldl1
-    , mkFoldr
-    , mkFoldrM
+    , foldl'
+    , foldlM'
+    , foldl1
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Fold.hs:24]: I am sorry, this had to be `foldl1'` instead. The same as we have in "Streamly.Prelude".

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**